core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/remotejobs/RemoteJob.java

```diff
@@ -44,21 +45,25 @@
 	private final String id;
 	private final ServiceProvider context;
 	private final Request<ServiceProvider, ?> request;
+	private final boolean autoClean;
```

Comment:
Another option would be to remove the `final` modifier from the `autoClean` field and set it to `false` in case of an error, which would simplify the implementation of the `run` method significantly.